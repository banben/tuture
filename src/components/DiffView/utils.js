function computeOldLineNumber(change) {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change) {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change) {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
