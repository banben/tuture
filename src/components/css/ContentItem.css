.ContentItem .ant-btn {
  border: none;
  background-color: #f7f7fa;
}

.indicate-icon {
  width: 10px;
}
