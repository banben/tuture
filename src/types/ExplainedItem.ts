type ExplainType = 'pre' | 'post';
type EditMode = 'edit' | 'preview';

type ToolType =
  | 'b'
  | 'i'
  | 'h'
  | 'list'
  | 'numbered list'
  | 'blockquotes'
  | 'link'
  | 'img'
  | 'code'
  | 'block code';

export { ExplainType, EditMode, ToolType };
