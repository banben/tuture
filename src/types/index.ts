interface Diff {
  file: string;
  explain: string;
  collapse?: boolean;
}

interface Step {
  name: string;
  commit: string;
  explain: string;
  diff: Diff[];
}

interface Tuture {
  name: string;
  language: string;
  version: string;
  topics: string[];
  description: string;
  maintainer: string;
  steps: Step[];
}

interface Change {
  content: string;
  type: string;
  isInsert?: boolean;
  isNormal?: boolean;
  isDelete?: boolean;
  lineNumber?: number;
  oldLineNumber?: number;
  newLineNumber?: number;
}

interface Hunk {
  content: string;
  newLines: number;
  oldLines: number;
  oldStart: number;
  newStart: number;
  changes: Change[];
  isPlain?: boolean;
}

export {
  Diff,
  Step,
  Tuture,
  Change,
  Hunk,
}
