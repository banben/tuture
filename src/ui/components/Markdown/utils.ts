const spliceStr = (str = '', typeStr: string, start: number, end: number) => {
  return str
    .slice(0, start)
    .concat(
      typeStr,
      str.slice(start, end),
      typeStr,
      str.slice(end, str.length),
    );
};

const insertStr = (str = '', typeStr: string, insertPosition: number) => {
  return str
    .slice(0, insertPosition)
    .concat(typeStr, str.slice(insertPosition, str.length));
};

export { spliceStr, insertStr };
