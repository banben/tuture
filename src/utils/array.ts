const isArray = (arr: any): boolean => {
  return Array.isArray(arr);
};

export {
  isArray,
};
