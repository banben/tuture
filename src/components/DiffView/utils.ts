import { Change as ChangeType } from '../../types';

function computeOldLineNumber(change: ChangeType): number {
  if (change.isInsert) {
    return -1;
  }

  return change.isNormal ? change.oldLineNumber : change.lineNumber;
}

function computeNewLineNumber(change: ChangeType): number {
  if (change.isDelete) {
    return -1;
  }

  return change.isNormal ? change.newLineNumber : change.lineNumber;
}

function getChangeKey(change: ChangeType): string {
  if (!change) {
    throw new Error('change is not provided');
  }

  const {
    isNormal,
    isInsert,
    lineNumber,
    oldLineNumber,
  } = change;

  if (isNormal) {
    return 'N' + oldLineNumber;
  }

  const prefix = isInsert ? 'I' : 'D';
  return prefix + lineNumber;
}

export {
  computeOldLineNumber,
  computeNewLineNumber,
  getChangeKey,
}
