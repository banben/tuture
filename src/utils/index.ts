import * as obj from './object';
import * as arr from './array';

const tutureUtilities = {
  ...arr,
  ...obj,
};

export default tutureUtilities;
