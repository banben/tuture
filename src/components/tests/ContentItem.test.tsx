import React from 'react';
import renderer from 'react-test-renderer';

import ContentItem from '../ContentItem';
import tutureUtilities from '../../utils/index';
import { Diff } from '../../types/index';
import { diff } from './utils/hunkData';

test('ContentItem render successfully', () => {
  const renderExplain = (explain: string[] | string): React.ReactNode | React.ReactNodeArray => {
    if (tutureUtilities.isArray(explain)) {
      const arrExplain = explain as string[];
      return (
        arrExplain.map((explainItem: string, i: number) =>
          <p key={i}>{explainItem}</p>)
      );
    }

    return <p>{explain}</p>;
  }

  const component = renderer.create(
    <ContentItem
      diff={diff}
      viewType={'unified'}
      commit={'1bd58a3'}
      renderExplain={renderExplain}
    />
  );
  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});
