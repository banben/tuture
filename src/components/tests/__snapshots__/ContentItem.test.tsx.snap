// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test ContentItem render successfully 1`] = `
<div
  className="ContentItem"
/>
`;
